vir/src/context.rs

```diff
@@ -484,16 +484,25 @@ impl<'tcx> VirCtxt<'tcx> {
         })
     }
 
-    const fn get_int_data(ty: Type, rust_ty: &ty::TyKind) -> (u32, bool) {
-        assert!(matches!(rust_ty, ty::Int(_) | ty::Uint(_)));
-        let TypeData::Int { bit_width, signed } = *ty else {
-            unreachable!();
-        };
-        assert!(matches!(rust_ty, ty::Int(_)) == signed);
-        (bit_width as u32, signed)
+    const fn get_int_data(rust_ty: &ty::TyKind) -> (u32, bool) {
+        match rust_ty {
+            ty::Int(ty::IntTy::Isize) => ((std::mem::size_of::<isize>() * 8) as u32, true),
```

Comment:
IIRC there is a better way to do this in the compiler APIs (@JonasAlaif?). Additionally, `isize` and `usize` should not use the pointer size on the current machine, but rather the one being targeted (also something available in `rustc_abi`).

Reply:
I updated this to use the target architecture; since `get_int_data` is a const fn, it's using [`target_pointer_width`](https://doc.rust-lang.org/reference/conditional-compilation.html#target_pointer_width) cfg var.